base/multidimensional.jl

```diff
@@ -367,6 +367,14 @@ module IteratorsMD
         return valid, (start[1], I...)
     end
 
+    function showarg(io::IO, c::CartesianIndices, toplevel)
+        if toplevel
+            print(io, "CartesianIndices ", UnitRange.(c.indices))
```

Comment:
Other users of `showarg` would print this as a function call, i.e., `CartesianIndices(1:3, 2:4)`. `showarg` came from a package called `ShowItLikeYouBuildIt`, so its output mimics how you construct the object itself.

It's worth thinking through the UnitRange carefully, for example:

```julia
julia> CartesianIndices(Base.IdentityUnitRange(-1:1))
CartesianIndices{1,Tuple{Base.IdentityUnitRange{UnitRange{Int64}}}} with indices -1:1:
 CartesianIndex(-1,)
 CartesianIndex(0,) 
 CartesianIndex(1,) 
```
It might be fine to do this if the `with indices -1:1` is preserved (which I think it will be).